packages/twenty-front/src/modules/object-record/record-index/options/hooks/useRecordData.ts

```diff
@@ -76,69 +77,63 @@ export const useTableData = ({
     (column) => column.metadata.fieldName === kanbanFieldMetadataName,
   );
   const columns = useRecoilValue(visibleTableColumnsSelector());
-  const selectedRowIds = useRecoilValue(selectedRowIdsSelector());
 
-  const hasUserSelectedAllRows = useRecoilValue(hasUserSelectedAllRowsState);
-  const tableRowIds = useRecoilValue(tableRowIdsState);
+  const contextStoreTargetedRecords = useRecoilValue(
+    contextStoreTargetedRecordsState,
+  );
+
+  const contextStoreCurrentObjectMetadataId = useRecoilValue(
+    contextStoreCurrentObjectMetadataIdState,
+  );
 
-  // user has checked select all and then unselected some rows
-  const userHasUnselectedSomeRows =
-    hasUserSelectedAllRows && selectedRowIds.length < tableRowIds.length;
+  const contextStoreTargetedRecordsFilters = useRecoilValue(
+    contextStoreTargetedRecordsFiltersState,
+  );
 
-  const hasSelectedRows =
-    selectedRowIds.length > 0 &&
-    !(hasUserSelectedAllRows && selectedRowIds.length === tableRowIds.length);
+  const { objectMetadataItem } = useObjectMetadataItemById({
+    objectId: contextStoreCurrentObjectMetadataId,
+  });
 
-  const unselectedRowIds = useMemo(
-    () =>
-      userHasUnselectedSomeRows
-        ? tableRowIds.filter((id) => !selectedRowIds.includes(id))
-        : [],
-    [userHasUnselectedSomeRows, tableRowIds, selectedRowIds],
+  const queryFilter = turnFiltersIntoQueryFilter(
+    contextStoreTargetedRecordsFilters,
+    objectMetadataItem?.fields ?? [],
   );
 
+  const { selectedRecordIds, excludedRecordIds } = contextStoreTargetedRecords;
+
   const findManyRecordsParams = useFindManyParams(
     objectNameSingular,
     recordIndexId,
   );
 
-  const selectedFindManyParams = {
-    ...findManyRecordsParams,
-    filter: {
-      ...findManyRecordsParams.filter,
-      id: {
-        in: selectedRowIds,
-      },
-    },
-  };
-
-  const unselectedFindManyParams = {
-    ...findManyRecordsParams,
-    filter: {
-      ...findManyRecordsParams.filter,
-      not: {
-        id: {
-          in: unselectedRowIds,
-        },
-      },
-    },
-  };
-
-  const usedFindManyParams =
-    hasSelectedRows && !userHasUnselectedSomeRows
-      ? selectedFindManyParams
-      : userHasUnselectedSomeRows
-        ? unselectedFindManyParams
-        : findManyRecordsParams;
-
   const {
     findManyRecords,
     totalCount,
     records,
     fetchMoreRecordsWithPagination,
     loading,
   } = useLazyFindManyRecords({
-    ...usedFindManyParams,
+    ...findManyRecordsParams,
+    filter: makeAndFilterVariables([
+      queryFilter,
+      selectedRecordIds !== 'all'
```

Comment:
nit: not a big fan of ternary-ception 😅 Up to you but I kind of agree with greptile there

Reply:
+1